org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/internal/DebugSession.java

```diff
@@ -0,0 +1,81 @@
+/*******************************************************************************
+* Copyright (c) 2017 Microsoft Corporation and others.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Eclipse Public License v1.0
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v10.html
+*
+* Contributors:
+*     Microsoft Corporation - initial API and implementation
+*******************************************************************************/
+
+package org.eclipse.jdt.ls.debug.internal;
+
+import java.util.List;
+
+import org.eclipse.jdt.ls.debug.IBreakpoint;
+import org.eclipse.jdt.ls.debug.IDebugSession;
+import org.eclipse.jdt.ls.debug.IEventHub;
+
+import com.sun.jdi.ThreadReference;
+import com.sun.jdi.VirtualMachine;
+
+public class DebugSession implements IDebugSession {
+    private VirtualMachine vm;
+    private EventHub eventHub = new EventHub();
+
+    public DebugSession(VirtualMachine virtualMachine) {
+        vm = virtualMachine;
+    }
+
+    @Override
+    public void start() {
+        eventHub.start(vm);
```

Comment:
Need do some initialization jobs, such as register ThreadStartRequest and ThreadDeathRequest to vm.